### PR TITLE
feat: use @types instead of typings

### PR DESCRIPTION
**Pull request by filipesilva**
See https://github.com/angular/angular-cli/issues/1356

At the moment ATL seems to have some problems loading types that are not declared explicitely in tsconfig, so they have to be added there.

### CODE REVIEW

File: addon/ng2/utilities/ts-path-mappings-webpack-plugin.ts [hunk 1]
```diff
@@ -1,252 +0,0 @@
-import * as path from 'path';
```
Comment (hansl) [inline addon/ng2/utilities/ts-path-mappings-webpack-plugin.ts]: Why is this file not needed anymore?

File: addon/ng2/blueprints/ng2/files/__path__/tsconfig.json [hunk 2]
```diff
@@ -9,6 +9,12 @@
     "moduleResolution": "node",
     "outDir": "<%= relativeRootPath %>/dist/out-tsc",
     "sourceMap": true,
-    "target": "es5"
+    "target": "es5",
+    "typeRoots": [
+      "../node_modules/@types"
+    ],
+    "types": [
```
Comment (hansl) [inline addon/ng2/blueprints/ng2/files/__path__/tsconfig.json]: So we'll have to have a discussion about this, and I need to talk to a bunch of other people. I'm against having jasmine typings for the _whole_ application. It should be for spec files only. But for that we would need to have 2 tsconfig files (or having different config). This was part of the discussions with the TypeScript team, and a solution does not exist yet. There are things that can be done though.

File: addon/ng2/utilities/ts-path-mappings-webpack-plugin.ts (hunk 1 above)
Comment (filipesilva) [inline addon/ng2/utilities/ts-path-mappings-webpack-plugin.ts]: It was the result of us waiting for a PR to be accepted on ATL. PR is in, and I had to update the ATL version, so I removed it.

File: addon/ng2/blueprints/ng2/files/__path__/tsconfig.json (hunk 2 above)
Comment (filipesilva) [inline addon/ng2/blueprints/ng2/files/__path__/tsconfig.json]: We can just add the ref to the tests I suppose. But our webpack pipeline can use different tsconfig as well, so we can do that.

**Comment by hansl** [comment]
LGTM. Let's get this in as is and have that discussion in a separate issue.

**Pull request merged**